pkgs/development/python-modules/dxpy/default.nix

```
@@ -0,0 +1,50 @@
+{
+  lib,
+  buildPythonPackage,
+  pythonOlder,
+  fetchgit,
+
+  # runtime dependencies
+  certifi,
+  urllib3,
+  python-dateutil,
+  websocket-client,
+  argcomplete,
+  psutil,
+}:
+
+buildPythonPackage rec {
+  pname = "dxpy";
+  version = "0.377.0";
+
+  disabled = pythonOlder "3.8";
+
+  src = "${fetchgit {
+    url = "https://github.com/dnanexus/dx-toolkit.git";
+    rev = "v0.377.0";
+    hash = "sha256-xcUSl5C9XKFc0TQh7wJPeTP7syZCwU5BWo55mtSBdN0=";
+  }}/src/python";
```

```suggestion
  src = fetchFromGitHub {
    owner = "dnanexus";
    repo = "dx-toolkit";
    rev = "v${version}";
    hash = "sha256-xcUSl5C9XKFc0TQh7wJPeTP7syZCwU5BWo55mtSBdN0=";
  };
  
  sourceRoot = "${src.name}/src/python";
```

https://github.com/NixOS/nixpkgs/blob/master/doc/stdenv/stdenv.chapter.md#sourceroot-var-stdenv-sourceroot